Packet++/src/SomeIpSdLayer.cpp

```diff
@@ -636,11 +636,25 @@ bool SomeIpSdLayer::isDataValid(const uint8_t* data, size_t dataLen)
 	if (!data ||
 		dataLen < sizeof(someipsdhdr) + sizeof(uint32_t) ||
 		dataLen < sizeof(someipsdhdr) + sizeof(uint32_t) + getLenEntries(data) + sizeof(uint32_t) ||
-		dataLen < be32toh(*((uint32_t *)(data + sizeof(someipsdhdr) + sizeof(uint32_t) + getLenEntries(data)))))
+		dataLen < sizeof(someipsdhdr) + sizeof(uint32_t) + getLenEntries(data) + sizeof(uint32_t) + getLenOptions(data))
 	{
 		return false;
 	}
 
+	size_t offsetOption = sizeof(someipsdhdr) + sizeof(uint32_t) + getLenEntries(data) + sizeof(uint32_t);
+	size_t lenOptions = getLenOptions(data);
+	uint32_t len = 0;
+	while (len < lenOptions)
+	{
+		if (len + sizeof(uint16_t) + 3 * sizeof(uint8_t) > lenOptions)
+			return false;
+
+		uint32_t lenOption = be16toh(*((uint16_t *)(data + offsetOption + len))) + 3 * sizeof(uint8_t);
+		len += lenOption;
+		if (len > lenOptions) // the last must be equal to lenOptions
+			return false;
+	}
```

Comment:
it's a bit weird to parse all the options in `isDataValid()`. Maybe we should fix `countOptions()` and `findOption()` to avoid going into non-allocated memory?